source/developers/how-tos/get-ready-for-python-dev.rst

```diff
@@ -0,0 +1,82 @@
+   person who is following the How-to.  The assumptions may link to other
+   how-tos if possible.
+
+* You have Git installed
+* You have Python 3.8 installed
+
+Steps
+*****
+
+.. note::
+
+   Some repositories will have a variation of these instructions in their README. Often it will be better to follow the instructions here instead, as the README instructions may be outdated, but keep an eye out for repositories that document special workflows.
+
+First-Time Setup
+================
+
+First, clone the repository with ``git clone <git_url>``. It's best to place all of your Open edX repositories in the same parent directory as each other, as a few of them assume that they will be arranged this way. (If you're not using devstack, this may not matter.)
+
+After navigating to the new directory in your shell, you'll need to set up a virtualenv and install requirements.
+
+A virtualenv contains a copy of your repo's Python dependencies. This avoids dependency version conflicts with your system packages and other repos.
+
+The most basic way of creating a virtualenv is to use the builtin ``venv`` module in Python::
+
+  python3.8 -m venv .venv-3.8
+
+This example creates a virtualenv at ``./.venv-3.8/``. Each time you will be working with Python in the repo, you'll need to activate the virtualenv for that terminal session::
+
+  source .venv-3.8/bin/activate
+
+Once you have done this, any calls to ``python``, ``pip``, or other Python executables will refer to the ones managed by the virtualenv.
+
+.. note::
+
+   If you create virtualenvs inside repos, you'll need to tell git to ignore them. The easiest way to do this is to create a file called ``.gitignore-global`` in your home directory and add the line ``.venv-3.*/``. Alternatively, you can create the virtualenvs elsewhere in your filesystem.
+
+Many developers use wrapper scripts (or write their own using shell aliases). One commonly used tool is ``virtualenvwrapper``, which manages the virtualenv directories outside of repositories; this avoids several issues with git and other tools being able to see the virtualenv, but will require explicitly naming each virtualenv.
+
+Working on a Repo
+=================
+
+#. Make sure you are on the default branch and have the latest version of the code::
+
+     git checkout main
+     git pull
+
+   Some repos instead have ``master`` as their default branch.
+
+#. Activate your virtualenv, as described above.
+#. Install or update the Python requirements for the repo into your virtualenv::
+
+     make requirements
+
+   In a few repos, this might also install NodeJS requirements, or the Makefile target might have a different name.
+
+#. Optionally, ensure you can run tests::
+
+     make test
+
+   Depending on the repository this might run not just unit tests, but also linters and other quality checks. The Makefile may also define ``make quality`` for code quality checks, or even a comprehensive ``test-all`` target.
```

Comment:
Gah, I would love to standardize on something. I've become quite partial to tutor's way, which is to [namespace all the checks under `make test-*` and use `make test` as the omni-check](https://github.com/overhangio/tutor/blob/7d1a6fff6e71e8df4d159cf97bdca4d799dbe90e/Makefile#L29-L49), but I think I'd be happy with any standard at this point.

I know that's outside the scope. I think how you described the current situation here is great.